### PR TITLE
chart: fix nesting of redis password placeholder

### PR DESCRIPTION
**Pull request by nadiamoe**
Previously, the redis password placeholder was not correctly indented, making it a noop. This causes the redis password to be rotated every time* the chart is rendered.

It is possible that this hasn't caused any outstanding issues so far for two reasons:
- The redis secret is correctly referenced by both mastodon and redis pods, and it is not persisted. This means if a new password is generated, restarting mastodon and redis pods will temporarily fix the issue.
- If `helm upgrade` is run and helm can connect to the cluster in templating time, the redis password is not being actually recreated, as the redis chart has code to retrieve the current password and use it instead: https://github.com/bitnami/charts/blob/21e1cdf237af351efb1c9cb5aabf4af9576692bd/bitnami/redis/templates/_helpers.tpl#L237

This however can cause issues in gitops environments, or if `helm template` is used. This PR simply moves the placeholder so the user-defined password is properly honored by the redis chart.

## Upgrade instructions

Set the new key `redis.auth.password` to a fixed value, e.g. the value that the incorrect `redis.password` key had, and run `helm upgrade`. The redis pod will restart and start using the new password. Mastodon pods pick up the new password when they are restarted, which should happen upon upgrade as well.

### CODE REVIEW

**Comment by roobre** [comment]
Rebased `main`.

**Comment by deepy** [comment]
I think either solution should work
```global.redis.password | Global Redis® password (overrides auth.password)```

**Comment by roobre** [comment]
@deepy Seems like at some point, the non-global key was also simply `password`, and bitnami moved it to `auth.password` but left the global untouched.

I don't get though what you mean by "either solution should work". Mastodon chart is not setting the `global.`, right? It's overriding the local (`redis.password` before this PR).

File: chart/values.yaml [hunk 1]
```diff
@@ -255,12 +255,12 @@ postgresql:
...
 # https://github.com/bitnami/charts/tree/master/bitnami/redis#parameters
 redis:
-  # -- you must set a password; the password generated by the redis chart will be
-  # rotated on each upgrade:
-  password: ""
-  # you can also specify the name of an existing Secret
-  # with a key of redis-password set to the password you want
-  # auth:
+  auth:
+    # you must set a password; the password generated by the redis chart will be
```
Comment (deepy) [inline chart/values.yaml:259]: For the sake of [helm-docs](https://github.com/norwoodj/helm-docs#valuesyaml-metadata) it'd be nice to retain the `--` so they continue to render like before

**Review by deepy — APPROVED** [review]

File: chart/values.yaml (hunk 1 above)
Comment (roobre) [reply chart/values.yaml]: Makes sense, should be fixed now!

**Comment by Gargron** [comment]
Moving the Helm chart out to mastodon/chart

**Pull request closed without merging**